### PR TITLE
Create UI for overriding expiration date

### PR DESCRIPTION
**Pull request by jforgue**

### CODE REVIEW

File: app/controllers/passengers_controller.rb [hunk 1]
```diff
@@ -40,6 +41,7 @@ def create
 
   def update
     if @passenger.update(passenger_params)
+      @passenger.doctors_note.update_attributes(overridden_by: @current_user)
```
Comment (dfaulken) [inline app/controllers/passengers_controller.rb]: We don't want to do this every time someone saves a passenger, right? Only when someone's specifically in the process of overriding the doctor's note.

File: app/helpers/passengers_helper.rb [hunk 2]
```diff
@@ -26,4 +26,8 @@ def passengers_table_class
       'row-border dispatch-table'
     end
   end
+
+  def passengers_form
```
Comment (dfaulken) [inline app/helpers/passengers_helper.rb]: It'd be nice if this method indicates that what it returns is a CSS class.

File: app/models/doctors_note.rb [hunk 3]
```diff
@@ -2,6 +2,13 @@
 
 class DoctorsNote < ApplicationRecord
   belongs_to :passenger
+  belongs_to :overridden_by, class_name: 'User', required: false
+  before_save :check_override
+
+  def check_override
```
Comment (dfaulken) [inline app/models/doctors_note.rb]: Should be a private method.

File: app/views/passengers/_form.html.haml [hunk 4]
```diff
@@ -48,9 +51,18 @@
     .field
       = d.label :expiration_date
       = d.text_field :expiration_date,
-        class: 'datepicker', id: :doctors_note_expiration_date
+        class: 'datepicker', id: 'doctors_note_expiration_date'
```
Comment (dfaulken) [inline app/views/passengers/_form.html.haml]: Why this change? Symbol should be fine here.

File: app/views/passengers/_form.html.haml [hunk 5]
```diff
@@ -48,9 +51,18 @@
...
+    .field
+      = d.label :override_expiration_date
```
Comment (dfaulken) [inline app/views/passengers/_form.html.haml]: `override_expiration_date` isn't the name of an attribute in the model, right?

File: app/views/passengers/_form.html.haml [hunk 6]
```diff
@@ -48,9 +51,18 @@
...
+      = d.check_box :override_expiration,
+        id: 'check-click'
+    - if @doctors_note.present?
+      .field#slide-date{class: passengers_form}
+        = d.label :enter_override_date
```
Comment (dfaulken) [inline app/views/passengers/_form.html.haml]: Same here. `enter_override_date` isn't the name of a model attribute.

File: app/views/passengers/_form.html.haml [hunk 7]
```diff
@@ -48,9 +51,18 @@
...
+        = d.text_field :override_until,
+          class: 'datepicker'
   .field
-    = form.label "Additional Notes"
+    = form.label :additional_notes
```
Comment (dfaulken) [inline app/views/passengers/_form.html.haml]: Same here too.

File: db/migrate/20170913210806_change_overridden_by_to_overridden_by_id.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+class ChangeOverriddenByToOverriddenById < ActiveRecord::Migration[5.1]
+  def change
+    rename_column :doctors_notes, :overridden_by, :overridden_by_id
```
Comment (dfaulken) [inline db/migrate/20170913210806_change_overridden_by_to_overridden_by_id.rb]: I disagree with this change. It's pretty common to have `_by` columns which point to users. Instead, you should tell the model that it's a foreign key. There are examples in our other apps.

File: spec/factories/doctors_notes.rb [hunk 9]
```diff
@@ -9,8 +9,7 @@
 
   trait :overridden do
     override_expiration true
-    # TODO: I think this should be defined as a real association.
-    overridden_by 1
+    overridden_by { create :user, :admin }
```
Comment (dfaulken) [inline spec/factories/doctors_notes.rb]: 👍

File: app/models/doctors_note.rb (hunk 3 above)
Comment (jforgue) [reply app/models/doctors_note.rb]: Right, I knew that.

File: app/views/passengers/_form.html.haml [hunk 10]
```diff
@@ -49,20 +49,20 @@
...
     - if @doctors_note.present?
-      .field#slide-date{class: passengers_form}
-        = d.label :enter_override_date
+      .field#slide-date{class: passengers_form_class}
+        = d.label 'enter_override_date'
         = d.text_field :override_until,
           class: 'datepicker'
   .field
-    = form.label :additional_notes
+    = form.label 'additional_notes'
```
Comment (dfaulken) [inline app/views/passengers/_form.html.haml]: No, symbols are correct here. They should match the names of the columns of the model.

Comment (dfaulken) [reply app/views/passengers/_form.html.haml]: In this case `:note`, and above `:override_expiration`

File: spec/helpers/passengers_helper_spec.rb [hunk 11]
```diff
@@ -21,7 +21,8 @@
     context 'will expire within warning period' do
       it "returns 'will_expire_soon'" do
         @doctors_note.update expiration_date: 5.days.since
-        expect(helper.passengers_table_row_class(@passenger)).to eql 'will_expire_soon'
+        expect(helper.passengers_table_row_class(@passenger))
```
[the remark this answers is not in the document]
Comment (mboneil10) [reply spec/helpers/passengers_helper_spec.rb]: Or did you write these tests?

File: app/views/passengers/_form.html.haml (hunk 10 above)
Comment (jforgue) [reply app/views/passengers/_form.html.haml]: Okay I see what you mean. I didn't realize you could change what would be displayed in the en.yml.

File: spec/helpers/passengers_helper_spec.rb (hunk 11 above)
[the remark this answers is not in the document]
Comment (jforgue) [reply spec/helpers/passengers_helper_spec.rb]: I only enforced rubocop on my changes and let rubocop -a fix what it could elsewhere.

[the remark this answers is not in the document]
Comment (mboneil10) [reply spec/helpers/passengers_helper_spec.rb]: Alright, no problem with a small app like this one. 👍

File: app/helpers/passengers_helper.rb [hunk 12]
```diff
@@ -26,4 +26,8 @@ def passengers_table_class
       'row-border dispatch-table'
     end
   end
+
+  def passengers_form_class
+    'hide_view' unless @doctors_note.override_expiration?
```
[the remark this answers is not in the document]
Comment (mboneil10) [reply app/helpers/passengers_helper.rb]: Is there a reason why you chose to do this here instead of in the HAML? Just wondering.

**Comment by jforgue** [comment]
Closes #48

File: spec/helpers/passengers_helper_spec.rb (hunk 11 above)
[the remark this answers is not in the document]
Comment (jforgue) [reply spec/helpers/passengers_helper_spec.rb]: I opened an issue to finish testing. 👍

File: app/helpers/passengers_helper.rb (hunk 12 above)
[the remark this answers is not in the document]
Comment (jforgue) [reply app/helpers/passengers_helper.rb]: I felt this kind of logic should stay out of the view and the helper was already there. 👍

**Comment by dfaulken** [comment]
Sorry for the delay on this one @jforgue. Great work.
